src/someio.cpp

```diff
@@ -0,0 +1,33 @@
+
+
+#include "someio.hpp"
+
+
+c_someio::c_someio()
+: m_stop(false)
+{ }
+
+void c_someio::stop_threadsafe() {
+	_info("someio sets m_stop");
+	m_stop=true;
+}
+
+void c_someio::set_sockopt_timeout(t_native_socket sys_handler, std::chrono::microseconds timeout) {
+	// ideas from http://stackoverflow.com/questions/2876024/linux-is-there-a-read-or-recv-from-socket-with-timeout
+	auto us = timeout.count();
+	struct timeval timespec;
+	timespec.tv_sec  = us / (1000*1000);
```

Comment:
that was for cancelio / setsocketopt. this is for linux.  This is ok (on linux)

How ever will need to do other version for windows probably